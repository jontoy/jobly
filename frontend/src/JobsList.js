import React from 'react';
import JobCard from './JobCard';

const JobsList = ({jobs, apply, unapply}) => {
    return (
        <ul className="JobsList list-unstyled">
            {jobs.map(({id, title, salary, equity, company_handle, state}) => 
            <li key={id}>
                <JobCard 
                    id={id} 
                    title={title} 
                    salary={salary} 
                    equity={equity} 
                    company_handle={company_handle} 
                    state={state} 
                    apply={apply} 
                    unapply={unapply}/>
            </li>)}
        </ul>
    )
}

export default JobsList;