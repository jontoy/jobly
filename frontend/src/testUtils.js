import React from "react";
import UserContext from "./userContext";

const demoUser = {
  username: "test",
  first_name: "test",
  last_name: "mctest",
  email: "test@test.net",
  photo_url: null,
  jobs:[]
};

const UserProvider = ({ children, currentUser = demoUser, appliedIds = [] }) => (
  <UserContext.Provider value={{currentUser, appliedIds}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
