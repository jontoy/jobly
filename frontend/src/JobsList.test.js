import React from 'react';
import { render } from '@testing-library/react';
import JobsList from './JobsList';
import {UserProvider} from './testUtils';

it('renders without crashing', () => {
    render(<JobsList jobs={[]} />);
});
  
it('matches snapshot', () => {
      const {asFragment} = render(<JobsList jobs={[]}/>);
    expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot with jobs data', () => {
    const jobs = [{id:1, title:'fake job', salary:140000, equity:0.01}]
    const {asFragment} = render(
    <UserProvider>
    <JobsList jobs={jobs}/>
    </UserProvider>
    );
  expect(asFragment()).toMatchSnapshot();
});

