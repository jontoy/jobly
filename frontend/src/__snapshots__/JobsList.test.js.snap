// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="JobsList list-unstyled"
  />
</DocumentFragment>
`;

exports[`matches snapshot with jobs data 1`] = `
<DocumentFragment>
  <ul
    class="JobsList list-unstyled"
  >
    <li>
      <div
        class="CompanyCard card p-2 mb-1"
      >
        <h5
          class="card-title"
        >
          fake job
        </h5>
        <p
          class="card-text"
        >
          <span>
            Salary: 140000
          </span>
          <br />
          <span>
            Equity: 0.01
          </span>
        </p>
        <button
          class="btn btn-success ml-auto"
        >
          Apply
        </button>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
